### DISCUSSION TITLE
`@strict` across includes

### DISCUSSION THREAD

**Discussion by vurvdev** [wiremod/wire#2915] [Suggestions]
Currently, if you `#include` a file, it will be `@strict` if the file including it is, even if the included file is not `@strict`.

This wasn't a problem before, but now that there are some `@strict`-specific warnings, it's created some confusion.
For example:

``foo.txt``
```golo
@strict

#include "bar"
```

``bar.txt``
```golo
if (1) {
    function test() {}
}
```

If you open ``bar.txt`` on its own, it will have no warnings. This is because legacy udfs are created at runtime.
But if you include it via `foo`, you'll get warnings from the include, as functions are static objects, and nesting them will do nothing.

## Solution

My thought for a solution was to simply properly have files be `@strict` with their own directives. I don't think it should leak into includes. This may cause some unforeseen issues though, so I am creating this discussion as an RFC

**Comment by deltamolfar** [comment]
I would make #include file remain strict, just ignore it's warnings. So avoidable runtime errors still would crash chips.
